Source/WebCore/html/HTMLFrameElementBase.cpp

```diff
@@ -100,17 +100,17 @@ void HTMLFrameElementBase::openURL(LockHistory lockHistory, LockBackForwardList
         frameName = getIdAttribute();
 
     auto completeURL = document().completeURL(m_frameURL);
-    auto finishOpeningURL = [this, weakThis = WeakPtr { *this }, frameName, lockHistory, lockBackForwardList, parentFrame = WTFMove(parentFrame), completeURL] {
+    auto finishOpeningURL = [weakThis = WeakPtr { *this }, frameName, lockHistory, lockBackForwardList, parentFrame = WTFMove(parentFrame), completeURL] {
         if (!weakThis)
             return;
-        Ref protectedThis { *this };
-        if (shouldLoadFrameLazily()) {
+        Ref protectedThis { *weakThis };
```

Comment:
A better pattern is to do this *before* the `if` check and check `if (!protectedThis)` above.